onnxruntime/core/providers/coreml/builders/impl/conv_op_builder.cc

```diff
@@ -46,18 +46,57 @@ Status ConvOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder, const N
   std::unique_ptr<COREML_SPEC::NeuralNetworkLayer> layer = CreateNNLayer(model_builder, node);
 
   const auto& input_defs = node.InputDefs();
+  const auto& output_defs = node.OutputDefs();
+  const auto& input_name = input_defs[0]->Name();
+  const auto& output_name = output_defs[0]->Name();
 
   const auto& weight_tensor = *model_builder.GetInitializerTensors().at(input_defs[1]->Name());
-  const auto& weight_shape = weight_tensor.dims();
+  auto weight_tensor_prime = weight_tensor;
+  auto& weight_shape = weight_tensor_prime.dims();
```

Comment:
Why do you need this weight_tensor_prime? You can just use the weight_tensor

CreateCoreMLWeight(,,,) does not actually look at the real shape of the  weight_tensor, it uses the dims() to calculate the number of elements, which does not change after add_dims(1)

Reply:
I am not too sure about the first question.. I think we still do need a copy for weight_tensor to make it mutable and as well as make weight_shape mutable?

But I updated the part for `CreateCoreMLWeight(...)` to just use weight_tensor and use common code.

Reply:
You can copy the weight shape out of the weight_tensor and modify it, it's not necessary to copy the entire weight tensor since the weight itself is unchanged

Reply:
okay. but if I copy `weight_shape` then the problem just goes back to having redundant code for `coreml_conv` etc, parts, where `weight_shape` gets used right? - because I will have something like `weight_shape_prime`

(I might have some misunderstanding here... you can clarify.)

And the weight_tensor_prime is only used with

Reply:
You can always copy the weight_tensor.dims() to a std::vector<int64_t> weight_shape{weight_tensor.dims().cbegin(),  weight_tensor.dims().cend()}, if it is 1d conv, append 1 to the end of weight_shape

Reply:
Thanks for the clarification. I see now. Updated.